File: python/redispy/simple.py
import redis
import sys


def main():
    if not (len(sys.argv) == 3 or len(sys.argv) == 4):
        print("Usage: host port password")
        exit(1)
    host = sys.argv[1]
    port = sys.argv[2]
    password = None
    if len(sys.argv) == 4:
        password = sys.argv[3]

    r = redis.StrictRedis(host=host, port=port, password=password)

    print("Set: {}".format(r.set("foo", "bar")))
    print("Get: {}".format(r.get("foo")))

    r.connection_pool.disconnect()


if __name__ == "__main__":
    main()

File: node.js/node-redis/simple.js

'use strict'
if (!(process.argv.length === 4 || process.argv.length === 5)) {
  console.log('Usage: host port password')
  process.exit(1)
}
const redis = require('redis')
var host = process.argv[2]
var port = process.argv[3]
var password = undefined
if (process.argv.length === 5) {
  password = process.argv[4]
}

const client = redis.createClient({ host: host, port: port, password: password })
client.on('error', function (error) {
  console.error('Error in connection')
  console.error(error)
})

client.set('foo', 'bar', function (err, reply) {
  if (err) {
    return console.error(err)
  }
  console.log('SET:', reply)
})

client.get('foo', function (err, reply) {
  if (err) {
    return console.error(err)
  }
  console.log('GET:', reply)
})

// close the connection after all commands are processed
client.quit()
